libtsuba/src/AddProperties.cpp

```diff
@@ -132,39 +134,44 @@ tsuba::AddPropertySlice(
   for (const tsuba::PropStorageInfo& prop : properties) {
     const std::string& name = prop.name;
     const katana::Uri& path = dir.Join(prop.path);
-    std::future<katana::Result<std::shared_ptr<arrow::Table>>> future =
-        std::async(
+    std::future<std::pair<
+        std::optional<katana::CopyableErrorInfo>,
+        std::shared_ptr<arrow::Table>>>
+        future = std::async(
```

Comment:
Yeah, it is a little ugly to have to return the error explicitly here, but by and large, this change is impressively contained.